sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/PartitionKeyInternal.java

```diff
@@ -170,6 +171,18 @@ public boolean equals(Object obj) {
         return this.compareTo(pki) == 0;
     }
 
+    @Override
+    public int hashCode() {
+        if (this.components == null || this.components.size() == 0) {
+            return 0;
+        }
+        int [] ordinals = new int[this.components.size()];
+        for (int i = 0; i < this.components.size(); i++) {
+            ordinals[i] = this.components.get(i).GetTypeOrdinal();
+        }
+        return Objects.hashCode(ordinals);
+    }
```

Comment:
Don't introduce more guava dependencies! :-) Use the standard Java APIs for hashcode

Reply:
:) removed

Reply:
Ok, now uses the default hashCode from the base class.

Reply:
and I see there is an issue link to complete the implementation.